hw-model/tests/model_tests.rs

```diff
@@ -271,3 +271,117 @@ fn test_pcr_extend() {
 
     model.step_until_exit_success().unwrap();
 }
+
+#[test]
+fn test_mbox_uc_response_dlen_after_data() {
+    let mut model = run_fw_elf_with_rand_puf(
+        &caliptra_builder::build_firmware_elf(&firmware::hw_model_tests::MAILBOX_RESPONDER)
+            .unwrap(),
+    );
+    let mbox = model.soc_mbox();
+    // Lock mailbox
+    assert!(!mbox.lock().read().lock());
+    assert!(mbox.lock().read().lock());
+
+    // This command writes 3 words to dint but doesn't update dlen immediately
+    mbox.cmd().write(|_| 0x1000_3000);
+    mbox.dlen().write(|_| 4);
+    mbox.datain().write(|_| 0x3a48_735b);
```

Comment:
is the goal of sending this word just to make sure it doesn't get included in the response? Might be good to comment that since otherwise it looks superfluous (the mbox responder doesn't use it).

Also I assume the choice of value is arbitrary. Is that correct?

Reply:
Yes. Fixed.